app/controllers/three_d_secure_controller.js

```diff
@@ -0,0 +1,121 @@
+
+// npm dependencies
+const _ = require('lodash')
+
+// local dependencies
+const responseRouter = require('../utils/response_router')
+const normalise = require('../services/normalise_charge')
+const paths = require('../paths')
+const {withAnalytics} = require('../utils/analytics')
+const connectorClient = require('../services/clients/connector_client')
+
+// constants
+const {views, threeDsEPDQResults} = require('../../config/charge_controller')
+const {CORRELATION_HEADER} = require('../../config/correlation_header')
+
+const routeFor = (resource, chargeId) => paths.generateRoute(`card.${resource}`, {chargeId: chargeId})
+
+const redirect = res => {
```

Comment:
I think we now have this redirect object in multiple places - we could extract it as a utility method. It does not have to be done as part of this PR, though.

Reply:
yep 👍  plan to do that and similar for other methods in another PR